lib/modules/versioning/debian/index.spec.ts

```diff
@@ -0,0 +1,430 @@
+    ${'slink'}        | ${false}
+    ${'potato'}       | ${false}
+    ${'woody'}        | ${false}
+    ${'sarge'}        | ${false}
+    ${'etch'}         | ${false}
+    ${'lenny'}        | ${false}
+    ${'squeeze'}      | ${false}
+    ${'wheezy'}       | ${false}
+    ${'jessie'}       | ${false}
+    ${'stretch'}      | ${true}
+    ${'buster'}       | ${true}
+    ${'bullseye'}     | ${true}
+    ${'bookworm'}     | ${false}
+    ${'trixie'}       | ${false}
+    ${'sid'}          | ${false}
+    ${'1.1'}          | ${false}
+    ${'1.2'}          | ${false}
+    ${'1.3'}          | ${false}
+    ${'2'}            | ${false}
+    ${'2.1'}          | ${false}
+    ${'2.2'}          | ${false}
+    ${'3'}            | ${false}
+    ${'4'}            | ${false}
+    ${'5'}            | ${false}
+    ${'6'}            | ${false}
+    ${'7'}            | ${false}
+    ${'8'}            | ${false}
+    ${'9'}            | ${true}
+    ${'10'}           | ${true}
+    ${'11'}           | ${true}
+    ${'12'}           | ${false}
+    ${'13'}           | ${false}
+    ${'sid'}          | ${false}
+    ${'experimental'} | ${false}
+    ${'stable'}       | ${true}
+    ${'oldstable'}    | ${true}
+    ${'oldoldstable'} | ${true}
+  `('isStable("$version") === $expected', ({ version, expected }) => {
+    spy.mockReturnValue(dt.valueOf());
+    expect(debian.isStable(version)).toBe(expected);
+  });
+
+  it.each`
+    version           | date            | expected
+    ${'7'}            | ${'2022-04-20'} | ${false}
+    ${'8'}            | ${'2022-04-20'} | ${false}
+    ${'9'}            | ${'2022-04-20'} | ${true}
+    ${'10'}           | ${'2022-04-20'} | ${true}
+    ${'11'}           | ${'2022-04-20'} | ${true}
+    ${'oldoldstable'} | ${'2022-04-20'} | ${true}
+    ${'oldstable'}    | ${'2022-04-20'} | ${true}
+    ${'stable'}       | ${'2022-04-20'} | ${true}
+    ${'7'}            | ${'2021-04-20'} | ${false}
+    ${'8'}            | ${'2021-04-20'} | ${false}
+    ${'9'}            | ${'2021-04-20'} | ${true}
+    ${'10'}           | ${'2021-04-20'} | ${true}
+    ${'11'}           | ${'2021-04-20'} | ${false}
+    ${'oldoldstable'} | ${'2021-04-20'} | ${false}
+    ${'oldstable'}    | ${'2021-04-20'} | ${true}
+    ${'stable'}       | ${'2021-04-20'} | ${true}
+    ${'7'}            | ${'2019-04-20'} | ${false}
+    ${'8'}            | ${'2019-04-20'} | ${true}
+    ${'9'}            | ${'2019-04-20'} | ${true}
+    ${'10'}           | ${'2019-04-20'} | ${false}
+    ${'11'}           | ${'2019-04-20'} | ${false}
+    ${'oldoldstable'} | ${'2019-04-20'} | ${false}
+    ${'oldstable'}    | ${'2019-04-20'} | ${true}
+    ${'stable'}       | ${'2019-04-20'} | ${true}
+  `(
+    'isStable("$version") @ "$date" === $expected',
+    ({ version, date, expected }) => {
+      const dt = DateTime.fromISO(date);
+      spy.mockReturnValue(dt.valueOf());
+      expect(debian.isStable(version)).toBe(expected);
+    }
+  );
```

Comment:
> I only like to have on small test like this to verify date handling for unintentional changes in future by some other author who doesn't know about long running renovate instances and module scope caching.
> 
> ```ts
> mockDate('future');
> expect(isStable('buster')).toBeFalse();
> mockDate('past');
> expect(isStable('buster')).toBeTrue();
> ```

doesnt this addresses the above?

Reply:
nope, as the time is fixed. so if someone else will dos some `improvement` and caches the computation again, this test won't fail, but my test will fail

Reply:
ive added the required test and reverted my last changes to the `rolling releases` class.
it will be refreshed once a day (on worst case).